### PR TITLE
Bugfix/issue 443

### PR DESCRIPTION
**Pull request by brettywhite**
## Issue:

Buffer methods called even if buffer was null.

## Fix:

Don't invoke buffer methods if null by surrounding calls in an else statement insuring the buffer is indeed not null

Also added the unit test file RegisteredAppTests that tests is the buffer is null or not. Before, the app would crash with a NPE and now it will not.

### CODE REVIEW

File: sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java [hunk 1]
```diff
@@ -2265,19 +2265,20 @@ public void handleMessage(int flags, byte[] packet){
 			}
 			if(buffer == null){
 				Log.e(TAG, "Unable to assemble message as buffer was null/not started");
```
Comment (joeygrover) [inline sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java]: I don't think we really need this log message. For the sake of clean code it should be removed and the this made into a single if statement, or null check if with a return

File: sdl_android/src/androidTest/java/com/smartdevicelink/transport/RegisteredAppTests.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+                // Call Handle Message
+                app.handleMessage(TransportConstants.BYTES_TO_SEND_FLAG_LARGE_PACKET_START,bytes);
+
+                // Insure that the buffer is not null, if it is the test will fail
+                assertNotNull(app.buffer);
+
+            }
+        });
+    }
```
Comment (brettywhite) [inline sdl_android/src/androidTest/java/com/smartdevicelink/transport/RegisteredAppTests.java]: see latest commit
